soc/xtensa/intel_adsp/common/include/cavs_hda.h

```diff
@@ -0,0 +1,303 @@
+#define HDA_HOST_OUT_BASE 0x72800
+#endif
+
+#ifndef HDA_HOST_IN_BASE
+#define HDA_HOST_IN_BASE 0x72c00
+#endif
+
+#ifndef HDA_STREAM_COUNT
+#define HDA_STREAM_COUNT 7
+#endif
+
+#ifndef HDA_REGBLOCK_SIZE
+#define HDA_REGBLOCK_SIZE 0x40
+#endif
+
+/* The read/write positions are masked to 24 bits */
+#define HDA_RWP_MASK 0x00FFFFFF
+
+/* Buffers must be 128 byte aligned, this mask enforces that */
+#define HDA_ALIGN_MASK 0xFFFFFF80
+
+
+#define HDA_ADDR(base, stream) (base + stream*HDA_REGBLOCK_SIZE)
+
+
+/* Gateway Control and Status Register */
+#define DGCS(base, stream) ((volatile uint32_t *)HDA_ADDR(base, stream))
+#define DGCS_SCS BIT(31) /* Sample container size */
+#define DGCS_GEN BIT(26) /* Gateway Enable */
+#define DGCS_L1ETP BIT(25) /* L1 Enter Prevent */
+#define DGCS_L1EXP BIT(25) /* L1 Exit Prevent */
+#define DGCS_FWCB BIT(23) /* Firmware Control Buffer */
+#define DGCS_GBUSY BIT(15) /* Gateway Busy */
+#define DGCS_TE BIT(14) /* Transfer Error */
+#define DGCS_BSC BIT(11) /* Buffer Segment Completion */
+#define DGCS_BOR BIT(10) /* Buffer Overrun */
+#define DGCS_BUR BIT(10) /* Buffer Underrun */
+#define DGCS_BF BIT(9) /* Buffer Full */
+#define DGCS_BNE BIT(8) /* Buffer Not Empty */
+#define DGCS_FIFORDY BIT(5) /* Enable FIFO */
+#define DGCS_BSCIE BIT(3) /* Buffer Segment Completion Interrupt Enable */
+
+/* Gateway Buffer Base Address */
+#define DGBBA(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x04))
+
+/* Gateway Buffer Size */
+#define DGBS(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x08))
+
+/* Gateway Buffer Pointer Increment */
+#define DGBFPI(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x0c))
+
+/* Gateway Buffer Read Pointer */
+#define DGBRP(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x10))
+
+/* Gateway Buffer Write Pointer */
+#define DGBWP(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x14))
+
+/* Gateway Buffer Segment Pointer */
+#define DGBSP(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x18))
+
+/* Gateway Minimum Buffer Size */
+#define DGMBS(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x1c))
+
+/* Gateway Linear Link Position Increment */
+#define DGLLPI(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x24))
+
+/* Gateway Linear Position In Buffer Increment */
+#define DGLPIBI(base, stream) ((volatile uint32_t *)(HDA_ADDR(base, stream) + 0x28))
+
+/**
+ * @brief Dump all the useful registers of an HDA stream to printk
+ *
+ * This can be invaluable when finding out why HDA isn't doing (or maybe is)
+ * doing what you want it to do. Macro so you get the file and line
+ * of the call site included.
+ *
+ * @param name String that contains a name of the hda stream (or anything really)
+ * @param base Base address of the IP register block
+ * @param sid Stream ID
+ */
+#define cavs_hda_dbg(name, base, sid)					\
+		printk("%s:%u %s(%u:0x%p), dgcs: 0x%x, dgbba 0x%x, "	\
```

Comment:
Should probably be LOG_DBG().  Standard guidelines these days seem to be that printk() usage shouldn't be commited to new code at the driver level or above, it's for arch code, tests (some platforms don't have logging available) and ad-hoc debugging.  I actually like this rule because it makes for an easy way to grep to see that I haven't left any stray debug stuff in a PR.

Also: it's a macro, it should be CAVS_HDA_DBG() or CAVS_HDA_REG_DUMP() or something.  I think the all caps rule is actually a MISRA thing?

Reply:
Our style guide says "like linux" and linux has lower case macro functions all over the place. Kind of ugly to all cap this, but if its a MISRA thing I suppose its one of those things.

Reply:
I looked through the MISRA docs, and our docs, and I didn't see anything that specifically says macros must be capped.